subscription-sender/internal/queue/consumer.go

```diff
@@ -0,0 +1,31 @@
+package queue
+
+import (
+	"context"
+
+	"github.com/nats-io/nats.go"
+
+	"github.com/RinaDish/subscription-sender/internal/services"
+	"github.com/RinaDish/subscription-sender/tools"
+)
+
+type NotificationService interface {
+	NotifySubscribers(ctx context.Context, notification services.Notification )
+}
+
+type Queue struct {
+	QueueConn *nats.Conn
+	subscriptionService NotificationService
+	subscriptionTopicName string
+	logger tools.Logger
+}
+
+
+func NewQueue(nats *nats.Conn, subscriptionTopicName string, subscriptionService NotificationService, logger tools.Logger) (*Queue) {
+	return &Queue{
+		QueueConn:     nats,
+		subscriptionService: subscriptionService,
+		subscriptionTopicName: subscriptionTopicName,
+		logger: logger.With("service", "queue"),
+	}
+}
```

Comment:
What if you will had other consume use cases, you will adjust this structure?

Maybe it's better to concretise what is this queue for, like 'NewSubscriptionNotifierConsumer' and give it proper package name?
So when you will need to add new subscriber you will do it in neighbour package

Reply:
Done, what do you mean "proper package name"?

Reply:
Put this like into queue/subscriptionNotifier/
So package name subscriptionNotifier for this